src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageApplicationModelProvider.cs

```diff
@@ -269,10 +275,17 @@ protected virtual PagePropertyModel CreatePropertyModel(PropertyInfo property)
             {
                 // Look for binding info on the handler if nothing is specified on the property.
                 // This allows BindProperty attributes on handlers to apply to properties.
-                var handlerType = property.DeclaringType;
-                var handlerAttributes = handlerType.GetCustomAttributes(inherit: true);
-                var handlerMetadata = _modelMetadataProvider.GetMetadataForType(property.DeclaringType);
-                bindingInfo = BindingInfo.GetBindingInfo(handlerAttributes, handlerMetadata);
+                var declaringType = property.DeclaringType;
+                var bindPropertyAttribute = declaringType.GetCustomAttribute<BindPropertiesAttribute>(inherit: true);
```

Comment:
Rinse and repeat my `bindPropertiesAttribute`, doc comments, and comments comments in `DefaultApplicationModelProvider`.